File: node/quickstart.js
// SnapAPI Node.js quickstart — turn https://numerixlabs.com into a PDF.
// Requires Node 18+ for built-in fetch. Docs: https://numerixlabs.com/docs

'use strict';

const fs = require('node:fs/promises');

const API_URL = 'https://api.numerixlabs.com/screenshot';

async function screenshot({ apiKey, url, format = 'pdf', width = 1280, out }) {
  const res = await fetch(API_URL, {
    method: 'POST',
    headers: {
      Authorization: `Bearer ${apiKey}`,
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ url, format, width }),
  });

  if (!res.ok) {
    throw new Error(`SnapAPI ${res.status}: ${await res.text()}`);
  }

  await fs.writeFile(out, Buffer.from(await res.arrayBuffer()));
  return out;
}

async function main() {
  const apiKey = process.env.SNAPAPI_KEY;
  if (!apiKey) {
    console.error('error: set SNAPAPI_KEY first (export SNAPAPI_KEY=sk_...)');
    console.error('       grab a free trial key at https://numerixlabs.com/trial');
    process.exit(1);
  }

  const out = await screenshot({
    apiKey,
    url: 'https://numerixlabs.com',
    format: 'pdf',
    out: 'numerixlabs.pdf',
  });

  console.log(`Saved ${out}`);
}

if (require.main === module) {
  main().catch((err) => {
    console.error(err.message);
    process.exit(1);
  });
}

module.exports = { screenshot };

File: python/quickstart.py
"""SnapAPI Python quickstart — turn https://numerixlabs.com into a PDF.

Docs: https://numerixlabs.com/docs
"""

import os
import sys

import requests

API_URL = "https://api.numerixlabs.com/screenshot"


def screenshot(api_key: str, url: str, out: str, *, fmt: str = "pdf", width: int = 1280) -> str:
    resp = requests.post(
        API_URL,
        headers={"Authorization": f"Bearer {api_key}"},
        json={"url": url, "format": fmt, "width": width},
        timeout=60,
    )
    resp.raise_for_status()
    with open(out, "wb") as f:
        f.write(resp.content)
    return out


def main() -> int:
    api_key = os.environ.get("SNAPAPI_KEY")
    if not api_key:
        print("error: set SNAPAPI_KEY first (export SNAPAPI_KEY=sk_...)", file=sys.stderr)
        print("       grab a free trial key at https://numerixlabs.com/trial", file=sys.stderr)
        return 1

    out = screenshot(api_key, "https://numerixlabs.com", "numerixlabs.pdf")
    print(f"Saved {out}")
    return 0


if __name__ == "__main__":
    sys.exit(main())


# --- No-dependency variant (standard library only) ---------------------------
# import json, os, urllib.request
#
# req = urllib.request.Request(
#     "https://api.numerixlabs.com/screenshot",
#     data=json.dumps({"url": "https://numerixlabs.com", "format": "pdf", "width": 1280}).encode(),
#     headers={
#         "Authorization": f"Bearer {os.environ['SNAPAPI_KEY']}",
#         "Content-Type": "application/json",
#     },
#     method="POST",
# )
# with urllib.request.urlopen(req, timeout=60) as resp, open("numerixlabs.pdf", "wb") as f:
#     f.write(resp.read())
